docs/proposal-007-apply.md

```diff
@@ -0,0 +1,137 @@
+# `eksctl apply`
+
+This proposal addresses a GitOps-compatible subcommand of `eksctl` called
+_tentatively_ `apply`. Its purpose is to subsume the various imperative commands
+`eksctl` provides now into one `apply` command that reconciles the actual state
+of an EKS cluster with the intended state as specified in the `eksctl` config file.
+
+This proposal is only concerned with the `apply` subcommand. Existing behavior
+and commands are unaffected. Continuing to support them shouldn't complicate
+the implementation too much as they are more or less subsets of
+operations performed by `apply`.
+
+## Questions to be answered
+
+-   Does the config represent the entire state of the config vs a partial
+    spec?
+    -   e.g. if I `apply` with a list of 1 nodegroups, are any other existing
+        nodegroups deleted?
+    -   do we also delete non-eksctl managed/created resources? i.e. does `eksctl`
+        own the cluster by default? (potential for configurability here)
```

Comment:
I don't think deleting would go well for us, but we will have find a new method of "adoption".

There would have to be a way to get an existing cluster's information and render it into a eksctl compatible config which then can be controlled via eksctl from then on.

It does get tricky if:
- Create non-eksctl cluster
- Get config, start controlling by eksctl
- Manually make "some change"
- What does eksctl do when it sees the new config? By then probably the same thing as it would if the cluster had been "owned" from the start.

_Originally posted by @Callisto13

Reply:
+1 being able to generate configs for unowned resources, I mentioned generating configs for imperatively created resources [here](https://github.com/weaveworks/eksctl-private/pull/233#discussion_r552489571). Expanding it to also get unowned is a great idea

_Originally posted by @aclevername_

Reply:
I have a vague memory of there being a risk here... didn't someone say in a call once that it is difficult (maybe impossible) to get a full cluster configuration?

_Originally posted by @Callisto13_